packages/relay/src/lib/eth.ts

```diff
@@ -419,9 +419,79 @@ export class EthImpl implements Eth {
+  /**
+   * Gets the value from a storage position at the given Ethereum address. 
+   *
+   * @param address
+   * @param slot
+   * @param blockNumberOrTag
+   */
+  async getStorageAt(address: string, slot: string, blockNumberOrTag?: string | null) : Promise<string> {
+    let result = EthImpl.zeroHex32Byte; // if contract or slot not found then return 32 byte 0
+    const blockEndTimestamp: string | undefined  = await this.getBlockEndTimeStamp(blockNumberOrTag);
+
+    // retrieve the timestamp of the contract
+    const contractResultsParams: IContractResultsParams = blockEndTimestamp 
+      ? { timestamp: `lte:${blockEndTimestamp}` } 
+      : {};
+    const limitOrderParams: ILimitOrderParams = { limit:1, order: 'desc' }; 
+    const contractResult = await this.mirrorNodeClient.getContractResultsByAddress(address, contractResultsParams, limitOrderParams);
+
+    if (contractResult?.results?.length > 0) {
+      // retrieve the contract result details 
+      await this.mirrorNodeClient.getContractResultsDetails(address, contractResult.results[0].timestamp)
+        .then( contractResultDetails => {
+          if(contractResultDetails && contractResultDetails.state_changes) {
+            // loop through the state changes to match slot and return value
+            for(const stateChange of contractResultDetails.state_changes) {
+              console.log("Hello");
+              if(stateChange.slot === slot) {
+                result = stateChange.value_written;
+              }
+            }
+          }
+        })
+        .catch( (e: any) => {
+          this.logger.error(
+            e,
+            'Failed to retrieve contract result details for contract address %s at timestamp=%s',
+            address,
+            contractResult.results[0].timestamp,
+          );
+          throw e;  
+        });
+    }
+
+    return result;
+  }
+
+  /**
+   * Gets the end timestamp of a block given a valid block number or 'earliest. 
+   * otherwise return undefined.
+   *
+   * @param blockNumberOrTag
+   */
+   private async getBlockEndTimeStamp(blockNumberOrTag?: string | null): Promise<string | undefined> {
+    let blockEndTimestamp: string | undefined;
+    // convert the block number into a timestamp if necessary
+    if (blockNumberOrTag && blockNumberOrTag !== 'latest' && blockNumberOrTag !== 'pending') {
+      let blockResponse: any;
+      if (blockNumberOrTag == 'earliest') {
+        blockResponse = await this.mirrorNodeClient.getBlock(0);
+      } else if (blockNumberOrTag.length < 32) {
+        // anything less than 32 characters is treated as a number
+        blockResponse = await this.mirrorNodeClient.getBlock(Number(blockNumberOrTag));
+      } else {
+        blockResponse = await this.mirrorNodeClient.getBlock(blockNumberOrTag);
+      }
+
+      if (_.isNil(blockResponse) || blockResponse.hash === undefined) {
+        // block not found. 
+        throw predefined.NO_SUITABLE_PEERS;
```

Comment:
I experimented on other test chains to see what errors I would get in for this error condition and replicated it here.

Reply:
I think that you caught this error because the Gateway that you accessed does not have nodes behind it that support the `eth_getStroageAt` query

Reply:
Per EIP-1474 it should be a resource not found error (error code -32001)

>If the block is not found, the callee SHOULD raise a JSON-RPC error (the recommended error code is -32001: Resource not found. If the tag is blockHash and requireCanonical is true, the callee SHOULD additionally raise a JSON-RPC error if the block is not in the canonical chain (the recommended error code is -32000: Invalid input and in any case should be different than the error code for the block not found case so that the caller can distinguish the cases). The block-not-found check SHOULD take precedence over the block-is-canonical check, so that if the block is not found the callee raises block-not-found rather than block-not-canonical. 
>
>[EIP-1474](https://eips.ethereum.org/EIPS/eip-1474#error-codes)